functions/go/starlark/starlark/processor.go

```diff
@@ -0,0 +1,43 @@
+package starlark
+
+import (
+	"fmt"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func Process(resourceList *fn.ResourceList) (bool, error) {
+	err := func() error {
+		fnCfg := resourceList.FunctionConfig
+		switch {
+		case fnCfg.IsGVK(configMapApiVersion, configMapKind):
+			cm := &ConfigMap{}
+			if err := cm.Config(resourceList.FunctionConfig); err != nil {
+				return err
+			}
+			return cm.Transform(resourceList)
+		case fnCfg.IsGVK(starlarkRunAPIVersion, starlarkRunKind):
+			sr := &StarlarkRun{}
+			if err := sr.Config(resourceList.FunctionConfig); err != nil {
+				return err
+			}
+			return sr.Transform(resourceList)
+		default:
+			return fmt.Errorf("`functionConfig` must be either %v or %v, but we got: %v",
+				schema.FromAPIVersionAndKind(configMapApiVersion, configMapKind).String(),
+				schema.FromAPIVersionAndKind(starlarkRunAPIVersion, starlarkRunKind).String(),
+				schema.FromAPIVersionAndKind(resourceList.FunctionConfig.GetAPIVersion(), resourceList.FunctionConfig.GetKind()).String())
+		}
```

Comment:
Some suggestions to simplify the logic and avoid code duplication (Config and Transform) and use the new KubeObject functions rather than `k8s.io/api/core/v1` or `kustomize/yaml` `Marshal`.
```go
type StarlarkFn struct {
     Name string
     Source string
     Params map[string]interface{}
}
func (s *StarlarkFn) Config(fn *fn.KubeObject) {
      s.Name = fnCfg.GetName()
      switch {
      case fnCfg.IsGVK(configMapApiVersion, configMapKind):
            s.Source = fnCfg.GetStringOrDie("data.source")
      case fnCfg.IsGVK(starlarkRunAPIVersion, starlarkRunKind):
            s.Source = fnCfg.GetStringOrDie("source")
            var params map[string]interface{}
            found, err = fnCfg.Get(&params, "params")
            s.Params = params
      default:
            error
      }  
}
func (s *StarlarkFn) Transform() {
    	return evaluateStarlarkToTransformResourceList(sr.Name, sr.Source, rl)
}
func Process() {
      err := func(){
             s := StarlarkFn{}
             s.Config(resourceList.FunctionConfig) 
             s.Transform(resourceList)    
      }
}
```

Reply:
then we don't need the `ConfigMap` and `StarlarkRun` struct, right?

Reply:
If we remove `StarlarkRun`, developers who want to do something programmatically with `StarlarkRun` type will need to define this type on their own. e.g. build another function that construct the `starlarkRun` functionConfig.

IMO when users work with types, I'd suggest using strongly typed object if possible since it can provide compile-time check. If strongly typed objects are not available, I'd suggest using unstructured approach next like `KubeObject` we have in the SDK.